src/v/cloud_storage/remote.cc

```diff
@@ -537,5 +537,73 @@ ss::future<download_result> remote::list_objects(
     _probe.failed_download();
     co_return download_result::timedout;
 }
+ss::future<download_result> remote::segment_exists(
+  const s3::bucket_name& bucket,
+  const remote_segment_path& segment_path,
+  retry_chain_node& parent) {
+    ss::gate::holder gh{_gate};
+    retry_chain_node fib(&parent);
+    retry_chain_logger ctxlog(cst_log, fib);
+    auto path = s3::object_key(segment_path());
+    auto [client, deleter] = co_await _pool.acquire();
+    auto permit = fib.retry();
+    vlog(ctxlog.debug, "Check segment {}", path);
+    std::optional<download_result> result;
+    while (!_gate.is_closed() && permit.is_allowed && !result) {
```

Comment:
nit: this retry pattern here seems similar to other functions. Can we encapsulate it (e.g. in a function that takes a lambda that creates a request)?